src/main/java/com/expediagroup/hiveberg/IcebergObjectInspectorGenerator.java

```diff
@@ -0,0 +1,82 @@
+package com.expediagroup.hiveberg;
+
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector;
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspectorFactory;
+import org.apache.hadoop.hive.serde2.objectinspector.primitive.PrimitiveObjectInspectorFactory;
+import org.apache.hadoop.hive.serde2.typeinfo.PrimitiveTypeInfo;
+import org.apache.hadoop.hive.serde2.typeinfo.TypeInfo;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class IcebergObjectInspectorGenerator {
+
+  private Schema schema;
+  private List<String> columnNames;
+  private List<TypeInfo> columnTypes;
+  private ObjectInspector oi;
+
+  public IcebergObjectInspectorGenerator(Schema schema) throws Exception {
+    IcebergSchemaToTypeInfo converter = new IcebergSchemaToTypeInfo();
+    this.schema = schema;
+    this.columnNames = setColumnNames();
+    this.columnTypes = converter.getColumnTypes(schema);
+  }
+
+  public List<String>  getColumnNames() {
+    return  columnNames;
+  }
+
+  public List<TypeInfo> getColumnTypes() {
+    return columnTypes;
+  }
+
+  public ObjectInspector getObjectInspector() {
+    return oi;
+  }
+
+  protected ObjectInspector createObjectInspector() throws Exception {
+    List<ObjectInspector> columnOIs = new ArrayList<>(columnNames.size());
+    for(int i = 0; i < columnNames.size(); i++) {
```

Comment:
```suggestion
    for(int i = 0; i < columnTypes.size(); i++) {
```
(I think it's clearer to use what you are iterating over as the index controller, less likely to introduce bugs if for some reason the other collection here has a different length)